src/lwpMediaDevices.js

```diff
@@ -1309,13 +1396,32 @@ export default class extends lwpRenderer {
   }
 
   _deviceKinds() {
-    return ["audiooutput", "audioinput", "videoinput"];
+    return ["ringoutput", "audiooutput", "audioinput", "videoinput"];
   }
 
   /** Shims */
 
-  _shimEnumerateDevices() {
-    return navigator.mediaDevices.enumerateDevices();
+  async _shimEnumerateDevices() {
+    const devices = await navigator.mediaDevices.enumerateDevices();
+    const outputDevices = devices.filter(
+      (device) => device.kind === "audiooutput"
+    );
+    let ringoutputDevices = [];
+
+    outputDevices.forEach((device) => {
+      ringoutputDevices.push(this._outputDeviceToRingDevice(device));
+    });
```

Comment:
I think you can optimize this a little bit more by doing something like this:
```js
const devices = await navigator.mediaDevices.enumerateDevices();
const ringoutputDevices = [];

devices.forEach((device) => {
  if (device.kind !== "audiooutput") return;
  ringoutputDevices.push(this._outputDeviceToRingDevice(device));
});
```

Reply:
Done ✅